lib/fluent/plugin/output.rb

```diff
@@ -800,7 +804,10 @@ def emit_sync(tag, es)
         @counter_mutex.synchronize{ @emit_count += 1 }
         begin
           process(tag, es)
-          @counter_mutex.synchronize{ @emit_records += es.size }
+          @counter_mutex.synchronize do
+            @emit_records += es.size
+            @emit_size += es.to_msgpack_stream.bytesize if @enable_size_metrics
```

Comment:
I considered more efficient way for this but I have no good idea for now.

Reply:
I see. I'm also considering what is the more efficient way.

Reply:
Please resolve the conflict and merge this.
We'll create a new pull request if we find a more efficient way.